### PR TITLE
page about asset compression

### PR DESCRIPTION
**Pull request by owaincuvelier**
@rmccue any feedback? not sure if I'm missing anything!

### CODE REVIEW

**Comment by rmccue** [comment]
I’d probably include this into the CDN page rather than a separate one, if it fits?

**Comment by owaincuvelier** [comment]
@rmccue I'd be in favour of splitting more things out into separate pages, this makes for much easier discoverability using search if the topic is stand-alone enough. Secondly, and this might just be semantics but compression will no longer exclusively occur at the CDN level.

It might be that it makes sense for this to be a subpage of the CDN though?

**Comment by rmccue** [comment]
Sure, subpage makes sense 👍 

Conceptually, I think the CDN and compression are related topics to users; whether it's technically implemented within the CDN I think is irrelevant.

**Comment by owaincuvelier** [comment]
@rmccue right yeah. I don't disagree. I've moved the page, happy with this as-is for now, if you are.

File: docs/cdn/asset-compression.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Asset Compression Strategy
+
+### What and why compress assets?
+
+For faster load speeds of your page, static assets are compressed at server, and then uncompressed by an end users browser. This is a contributing factor to SEO scores, but generally good practice to help reduce overall bandwidth usage.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:5]: "compressed at the server"
"uncompressed by a visitor's browser"

File: docs/cdn/asset-compression.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Asset Compression Strategy
+
+### What and why compress assets?
+
+For faster load speeds of your page, static assets are compressed at server, and then uncompressed by an end users browser. This is a contributing factor to SEO scores, but generally good practice to help reduce overall bandwidth usage. 
+
+Those static assets include .js or CSS files, images, and also the generated HTML files of your site.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:7]: ".js and CSS files"

File: docs/cdn/asset-compression.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### How do we compress assets?
+
+Assets are compressed using GZIP and is done via NGINX. When they’re sent as a response to a request they’ll also then be cached at the CDN for faster delivery of that assets to subsequent requests.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:11]: "that asset"

File: docs/cdn/asset-compression.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+The [MIME types](https://en.wikipedia.org/wiki/Media_type) we compress are listed below.
+
+`text/plain, text/css, text/html, application/json, application/javascript, application/x-javascript, text/xml, application/xml, application/xml+rss, text/javascript, application/x-font-ttf, font/opentype, application/vnd.ms-fontobject, image/svg+xml`
+
+Images are compressed and cached separately, as we leverage [Dynamic Image Resizing](https://docs.altis-dxp.com/media/dynamic-images/), the Images are created on the fly, and then cached at the CDN layer for serving the image. Images will attempt to be served via `[WebP](https://developers.google.com/speed/webp/)` format if the browser supports it, falling back to the image default format.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:17]: "if the browser supports [WebP](link...), images will be served in that format, falling..."

File: docs/cdn/asset-compression.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+### I’ve heard of Brotli compression, and doesn’t it offer better compression?
+
+It can provide a meaningful boost to compression for non-image assets like javascript or CSS. But support for Brotli compression is only partially supported.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:21]: "support for ... supported"?? Should this be "support for Brotli is only partially implemented" or "not supported by all browsers"? Or even "support for Brotli is still patchy/inconsistent"?

File: docs/cdn/asset-compression.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+If you’re do want to use Brotli for compression we can enable Brotli for on-the-fly compression at the CDN layer, but with the exception of the MIME type of `text/html` which will have to be compressed with GZIP, due to the dynamic nature of the HTML generation.
```
Comment (mikelittle) [inline docs/cdn/asset-compression.md:23]: "if you want..."
"...but not for the MIME type `text/html` which is dynamically generated at the server and compressed with `gzip`."

**Review by mikelittle — COMMENTED** [review]
Some comments...

**Comment by owaincuvelier** [comment]
@mikelittle thank you so much! I've commited some improvements!

File: docs/cdn/asset-compression.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# Asset Compression Strategy
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:1]: Suggest also renaming the file to `compression-optimisation.md`

```suggestion
# Compression & Optimisation
```

File: docs/cdn/asset-compression.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+### What and why compress assets?
+
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:4]: Levels are wrong here (should be `##`), but suggest removing this title entirely and combining with opening statement.

File: docs/cdn/asset-compression.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+For faster load speeds of your page, static assets are compressed at the server, and then uncompressed by a vistor's browser. This is a contributing factor to SEO scores, but generally good practice to help reduce overall bandwidth usage. 
+
+Those static assets include .js and CSS files, images, and also the generated HTML files of your site.
+
+### How do we compress assets?
+
+Assets are compressed using GZIP and is done via NGINX. When they’re sent as a response to a request that asset will also be cached at the CDN for faster delivery of that assets to subsequent requests.
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:11]:
```suggestion
Assets are compressed using GZIP for browsers which indicate support for it. This includes all modern browsers on desktop and mobile platforms. Like other responses, compressed assets are cached on the CDN for faster delivery to subsequent requests.
```

File: docs/cdn/asset-compression.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+Images are compressed and cached separately, as we leverage [Dynamic Image Resizing](https://docs.altis-dxp.com/media/dynamic-images/), the Images are created on the fly, and then cached at the CDN layer for serving the image. Images will attempt to be served via `[WebP](https://developers.google.com/speed/webp/)` format if the browser supports WebP, images will be served in that format, falling back to the image default format.
+
+### I’ve heard of Brotli compression, and doesn’t it offer better compression?
+
+It can provide a meaningful boost to compression for non-image assets like javascript or CSS. Brotli compression is only partially supported, meaning it can only be used on-the-fly by at the CDN, and not for `text/html` MIME Types and [not all browsers support Brotli](https://docs.w3cub.com/browser_support_tables/brotli).
+
+To enable Brotli for on-the-fly compression at the CDN layer (with the exception of the MIME type of `text/html` which will have to be compressed with GZIP), make a request to [Altis Support](https://docs.altis-dxp.com/guides/getting-help-with-altis/).
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:23]:
```suggestion
Brotli support on Altis is currently available on an opt-in basis. To enable Brotli on your application, contact [Altis Support](https://docs.altis-dxp.com/guides/getting-help-with-altis/).
```

File: docs/cdn/asset-compression.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+### Further reading:
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:25]: Suggest eliminating this section, and only having inline links.

**Review by rmccue — CHANGES_REQUESTED** [review]

File: docs/cdn/asset-compression.md (hunk 11 above)
Comment (rmccue) [reply docs/cdn/asset-compression.md:32]: I'd drop the whole further reading, yeah; we can link inline where relevant.

File: docs/cdn/asset-compression.md (hunk 8 above)
Comment (owaincuvelier) [reply docs/cdn/asset-compression.md]: Cool. Updated.

**Comment by rmccue** [comment]
@owaincuvelier Is this ready for re-review? Hit the 🔃 button whenever you're done

**Comment by owaincuvelier** [comment]
@rmccue i hit the button but nothing happened.

File: docs/cdn/asset-compression.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+Images are compressed, optimised, and cached separately, through support for [dynamic images](https://docs.altis-dxp.com/media/dynamic-images/). Optimised and resized images are created on the fly, and then cached at the CDN layer for a year.
+
+For browsers which support it, images will be automatically converted to [WebP](https://developers.google.com/speed/webp/) format.
+
+## What about Brotli compression?
+
+[Brotli](https://en.wikipedia.org/wiki/Brotli) is a modern compression algorithm developed by Google to replace gzip with higher compression ratios. [Support for Brotli is available in all modern browsers](https://caniuse.com/brotli), constituting ~96% of users.
+
+Brotli can provide a meaningful boost to compression for non-image assets like JavaScript or CSS. Altis has partial support for Brotli at the CDN layer. Support for dynamic (`text/html`) responses is not currently available.
```
Comment (rmccue) [inline docs/cdn/asset-compression.md:31]:
```suggestion
[Brotli](https://en.wikipedia.org/wiki/Brotli) is a modern compression algorithm developed by Google to replace gzip with higher compression ratios. [Support for Brotli is available in all modern browsers](https://caniuse.com/brotli), constituting ~96% of users. Brotli can provide a meaningful boost to compression for non-image assets like JavaScript or CSS.

Altis has partial support for Brotli at the CDN layer. Support for dynamic (`text/html`) responses is not currently available.
```

**Review by rmccue — APPROVED** [review]
One slight rebalancing of the paragraphs.

**Pull request merged**